misk-action-scopes/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -57,12 +57,14 @@ class ActionScope @Inject internal constructor(
   }
 
   /** Starts the scope on a thread with the provided seed data */
-  fun enter(seedData: Map<Key<*>, Any?>): ActionScope {
+  override fun enter(seedData: Map<out Any, Any?>): Scope {
+    val seedMap: Map<Key<*>, Any?> = seedData as Map<Key<*>, Any?>
```

Comment:
We might need to revisit this when we write the Armeria adapter that extends Scope, since not sure the assumption that Key class will be used.

Reply:
Agreed. For now, we are keeping the Guice module in misk-api.